lib/ProteinStructureUtils/Utils/PDBUtils.py

```diff
@@ -74,15 +98,103 @@ def _file_to_data(self, file_path):
+        #structure.header = structure.header
+        #print(f'The mmcif structure header for {cif}*************************:\n\n')
+        #kwds = structure.header.get('keywords')
+        #print(f'Keywords: {kwds}')
+        #resl = structure.header.get('resolution')
+        #print(f'Resolution: {resl}')
+
+        # create a dictionary that maps all mmCIF tags in an mmCIF file to their values
+        #mmcif_dict = MMCIF2Dict(cif)  # TypeError: 'module' object is not callable
+        #print("The mmcif dictionary:*****************************************\n\n")
+        # E.g., get the solvent content from an mmCIF file:
+        #sc = mmcif_dict["_exptl_crystal.density_percent_sol"]
+        #print(f'_exptl_crystal.density_percent_sol: {sc}')
+        # get the list of the y coordinates of all atoms
+        #y_list = mmcif_dict["_atom_site.Cartn_y"]
+        # print(f'_atom_site.Cartn_y: {y_list}')
+
+        ppb = PPBuilder()
+        pp_list = []
+        pp_no = 0
+        for pp in ppb.build_peptides(structure):
+            pp_no += 1
+            my_seq = pp.get_sequence()
+            pp_list += str(my_seq)
+        seq = ''.join(pp_list)
+
+        cpd_dict = dict()
+        cpd = structure.header.get('compound', {})
+        print(f'Compound:\n {cpd}')
+        if cpd and cpd.get('1'):
+            cpd_dict = cpd.get('1')
+
+        src_dict = dict()
+        src = structure.header.get('source', {})
+        if src and src.get('1'):
+            src_dict = src.get('1')
+
+        hd = self._upload_to_shock(file_path)
+
+        mmcif_data = {
+            'name': structure.header.get('name', ''),
+            'head': structure.header.get('head', ''),
+            'rcsb_id': structure.header.get('rcsb_id', ''),
+            'deposition_date': structure.header.get('deposition_date', ''),
+            'release_date': structure.header.get('release_date', ''),
+            'structure_method': structure.header.get('structure_method', ''),
+            'resolution': structure.header.get('resolution', 0.0),
+            'structure_reference': structure.header.get('structure_reference', []),
+            'keywords': structure.header.get('keywords', ''),
+            'author': structure.header.get('author', ''),
+            'compound': cpd_dict,
+            'source': src_dict,
+            'num_models': structure.header.get('num_models', 0),
+            'num_chains': structure.header.get('num_chains', 0),
+            'num_residues': structure.header.get('residues', 0),
+            'num_atoms': structure.header.get('num_atoms', 0),
+            'num_het_atoms': structure.header.get('num_het_atoms', 0),
+            'num_water_atoms': structure.header.get('num_water_atoms', 0),
+            'num_disordered_atoms': structure.header.get('num_disordered_atoms', 0),
+            'num_disordered_residues': structure.header.get('num_disordered_residues', 0),
+            'proteins': structure.header.get('proteins', []),
```

Comment:
You should check `structure.header.get('proteins')` does return required fields for `ProteinData`. I think it needs to include at least`id`, `sequence`, and `md5`. I don't think proteins can be an empty list.